addons/wowl/static/src/components/loading_indicator/loading_indicator.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, useState } from "@odoo/owl";
+import type { OdooEnv } from "../../types";
+
+export class LoadingIndicator extends Component<{}, OdooEnv> {
+  static template = "wowl.LoadingIndicator";
+  state = useState({
+    count: 0,
+    isShow: false,
+  });
+  rpcIds: Set<number> = new Set();
+  blockUITimer!: number;
+
+  constructor() {
+    super(...arguments);
+    this.env.bus.on("RPC:REQUEST", this, (rpcId: number) => {
+      this.requestCall(rpcId);
+    });
+    this.env.bus.on("RPC:RESPONSE", this, (rpcId: number) => {
+      this.responseCall(rpcId);
+    });
+  }
+
+  requestCall(rpcId: number) {
+    if (this.state.count === 0) {
+      this.state.isShow = true;
+      this.blockUITimer = setTimeout(() => {
+        this.env.services.ui.block();
+      }, 3000);
+    }
+    this.rpcIds.add(rpcId);
+    this.state.count = this.rpcIds.size;
```

Comment:
why not count++? Actually, why do you need a set?

Reply:
This is to avoid synchronization problems. Or if you don't catch an rpc. (it was a problem you told me about)

Reply:
I don't think so. All rpc requests are taken into account, and thus require the counter to be incremented (it isn't possible to have a new request with an idea that is already in the set)

Reply:
So I don't check the RPC id anymore ? I just increment or decrement the count ?